modules/puppet_proxy_puppet_api/plugin_configuration.rb

```diff
@@ -16,15 +16,35 @@ def load_classes
       require 'puppet_proxy_puppet_api/v3_api_request'
       require 'puppet_proxy_puppet_api/v3_environments_retriever'
       require 'puppet_proxy_puppet_api/v3_classes_retriever'
+      require 'puppet_proxy_puppet_api/v3_environmentclassesapi_classes_retriever' if RUBY_VERSION > '1.8.7'
     end
 
     def load_dependency_injection_wirings(container_instance, settings)
       container_instance.dependency :environment_retriever_impl,
                                     lambda {::Proxy::PuppetApi::V3EnvironmentsRetriever.new(settings[:puppet_url], settings[:puppet_ssl_ca], settings[:puppet_ssl_cert], settings[:puppet_ssl_key])}
 
-      container_instance.dependency :class_retriever_impl,
-                                    lambda {::Proxy::PuppetApi::V3ClassesRetriever.new(settings[:puppet_url], settings[:puppet_ssl_ca], settings[:puppet_ssl_cert], settings[:puppet_ssl_key])}
+      if settings[:puppet_version].to_s < "4.4" || RUBY_VERSION < '1.9.3'
+        container_instance.dependency :class_retriever_impl,
+                                      lambda {::Proxy::PuppetApi::V3ClassesRetriever.new(settings[:puppet_url], settings[:puppet_ssl_ca], settings[:puppet_ssl_cert], settings[:puppet_ssl_key])}
+        container_instance.dependency :class_cache_initializer, lambda {Proxy::PuppetApi::NoopClassesCacheInitializer.new}
+      end
 
+      if settings[:puppet_version].to_s >= "4.4" && RUBY_VERSION > '1.8.7'
```

Comment:
Prefer `else`, as this is just an inversion of the previous conditional I think.